### PR TITLE
added communication apps to application setup + small fix in dev tools

### PR DESCRIPTION
**Pull request by guruswarupa**
## Type of Change
- [x] New feature

## Description
Added communication apps into application-setup

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/applications-setup/communication-apps/discord-setup.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm discord 
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y https://download1.rpmfusion.org/nonfree/fedora/rpmfusion-nonfree-release-$(rpm -E %fedora).noarch.rpm
+                "$ESCALATION_TOOL" "$PACKAGER" install -y discord
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager. Please install Discord manually.${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/discord-setup.sh:24]:
```suggestion
                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
```

File: core/tabs/applications-setup/communication-apps/jitsi-meet-setup.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install jitsi
+                ;;
+            pacman)
+                "$AUR_HELPER" -S --needed --noconfirm jitsi-meet
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y jitsi-meet
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager. Please install Jitsi meet manually.${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/jitsi-meet-setup.sh:24]:
```suggestion
                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
```

File: core/tabs/applications-setup/communication-apps/signal-setup.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm signal-desktop
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" copr enable luminoso/Signal-Desktop
+                "$ESCALATION_TOOL" "$PACKAGER" install -y signal-desktop 
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager. Please install Signal manually.${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/signal-setup.sh:26]:
```suggestion
                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
```

File: core/tabs/applications-setup/communication-apps/telegram-setup.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install telegram-desktop
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm telegram-desktop 
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y telegram-desktop 
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager. Please install Telegram manually.${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/telegram-setup.sh:22]:
```suggestion
                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
```

File: core/tabs/applications-setup/communication-apps/thunderbird-setup.sh [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install thunderbird
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm thunderbird
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y thunderbird 
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager. Please install Thunderbird manually.${RC}"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/thunderbird-setup.sh:22]:
```suggestion
                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
```

**Review by adamperkowski — CHANGES_REQUESTED** [review]
Add `task_list`s to the toml

File: core/tabs/applications-setup/communication-apps/discord-setup.sh [hunk 6]
```diff
@@ -0,0 +1,35 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+installDiscord() {
+    if ! command_exists discord; then
+        printf "%b\n" "${YELLOW}Installing Discord...${RC}"
+        case "$PACKAGER" in
+            apt-get|nala)
+                curl -L -o discord.deb "https://discord.com/api/download?platform=linux&format=deb"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/discord-setup.sh:10]:
```suggestion
                curl -Lo discord.deb "https://discord.com/api/download?platform=linux&format=deb"
```

File: core/tabs/applications-setup/communication-apps/discord-setup.sh [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" install -y discord.deb
+                ;;
+            zypper)
+                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install discord
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm discord 
+                ;;
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/discord-setup.sh:18]: You need to set up `multilib`

File: core/tabs/applications-setup/communication-apps/jitsi-meet-setup.sh [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+            apt-get|nala)
+                curl https://download.jitsi.org/jitsi-key.gpg.key | "$ESCALATION_TOOL" sh -c 'gpg --dearmor > /usr/share/keyrings/jitsi-keyring.gpg'
+                printf "%b\n" 'deb [signed-by=/usr/share/keyrings/jitsi-keyring.gpg] https://download.jitsi.org stable/' | "$ESCALATION_TOOL" tee /etc/apt/sources.list.d/jitsi-stable.list > /dev/null
+                "$ESCALATION_TOOL" "$PACKAGER" -y install jitsi-meet
+                ;;
+            zypper)
+                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install jitsi
+                ;;
+            pacman)
+                "$AUR_HELPER" -S --needed --noconfirm jitsi-meet
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/jitsi-meet-setup.sh:18]:
```suggestion
                ;;
            pacman)
                "$AUR_HELPER" -S --needed --noconfirm jitsi-meet-bin
```

File: core/tabs/applications-setup/communication-apps/slack-setup.sh [hunk 9]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+installSlack() {
+    if ! command_exists slack; then
+        printf "%b\n" "${YELLOW}Installing Slack...${RC}"
+        case "$PACKAGER" in
+            pacman)
+                "$AUR_HELPER" -S --needed --noconfirm slack-desktop
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/slack-setup.sh:10]: https://aur.archlinux.org/packages/slack-desktop-wayland

File: core/tabs/applications-setup/communication-apps/zoom-setup.sh [hunk 10]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+installZoom() {
+    if ! command_exists signal; then
+        printf "%b\n" "${YELLOW}Installing Zoom...${RC}"
+        case "$PACKAGER" in
+            pacman)
+                "$AUR_HELPER" -S --needed --noconfirm zoom
```
Comment (adamperkowski) [inline core/tabs/applications-setup/communication-apps/zoom-setup.sh:10]: You have `--needed` here but not in other cases. Fix this.

**Review by adamperkowski — CHANGES_REQUESTED** [review]

**Comment by kleidis** [comment]
This is unrelated to the PR itself 

But i feel like these simple scripts that just download apps should be merged into one app_install script 

And a new horizontal tab should be added for just apps, so like WinUtil, basically

**Comment by guruswarupa** [comment]
> This is unrelated to the PR itself
> 
> But i feel like these simple scripts that just download apps should be merged into one app_install script
> 
> And a new horizontal tab should be added for just apps, so like WinUtil, basically

what do you mean by `unrelated`?

File: core/tabs/applications-setup/communication-apps/jitsi-setup.sh [hunk 11]
```diff
@@ -0,0 +1,36 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+installJitsi() {
+    if ! command_exists jitsi-meet; then
+        printf "%b\n" "${YELLOW}Installing Jitsi meet...${RC}"
+        case "$PACKAGER" in
+            apt-get|nala)
+                curl https://download.jitsi.org/jitsi-key.gpg.key | "$ESCALATION_TOOL" sh -c 'gpg --dearmor > /usr/share/keyrings/jitsi-keyring.gpg'
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/communication-apps/jitsi-setup.sh:10]: dont use sh -c

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/communication-apps/jitsi-setup.sh [hunk 12]
```diff
@@ -0,0 +1,36 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+installJitsi() {
+    if ! command_exists jitsi-meet; then
+        printf "%b\n" "${YELLOW}Installing Jitsi meet...${RC}"
+        case "$PACKAGER" in
+            apt-get|nala)
+                curl https://download.jitsi.org/jitsi-key.gpg.key | "$ESCALATION_TOOL" 'gpg --dearmor > /usr/share/keyrings/jitsi-keyring.gpg'
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/communication-apps/jitsi-setup.sh:10]:
```suggestion
                curl https://download.jitsi.org/jitsi-key.gpg.key | "$ESCALATION_TOOL" gpg --dearmor > /usr/share/keyrings/jitsi-keyring.gpg
```

File: core/tabs/applications-setup/communication-apps/zoom-setup.sh [hunk 13]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh -e
+
+. ../../common-script.sh
+
+installZoom() {
+    if ! command_exists signal; then
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/communication-apps/zoom-setup.sh:6]:
```suggestion
    if ! command_exists zoom; then
```

File: core/tabs/applications-setup/tab_data.toml [hunk 14]
```diff
@@ -45,6 +45,50 @@ description = "Ngrok is a tool that creates secure, temporary tunnels to expose
...
+[[data.entries]]
+name = "Install Telegram"
+description = "Telegram is a cloud-based messaging app known for its speed and security, offering features like group chats, channels, and end-to-end encrypted calls."
+script = "communication-apps/telegram-setup.sh"
+task_list = "installation"
+
+[[data.entries]]
+name = "Install Zoom"
+description = "Zoom is a widely-used video conferencing platform that allows users to host virtual meetings, webinars, and online collaboration with features like screen sharing and recording."
+script = "communication-apps/zoom-setup.sh"
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/tab_data.toml:84]:
```suggestion
script = "communication-apps/zoom-setup.sh"
task_list = "installation"
```

**Review by nnyyxxxx — APPROVED** [review]
looks decent

File: core/tabs/applications-setup/tab_data.toml [hunk 15]
```diff
@@ -45,6 +45,51 @@ description = "Ngrok is a tool that creates secure, temporary tunnels to expose
...
+task_list = "installation"
+
+[[data.entries]]
+name = "Install Thunderbird"
```
Comment (adamperkowski) [inline core/tabs/applications-setup/tab_data.toml:88]:
```suggestion
name = "Discord"
description = "Discord is a versatile communication platform for gamers and communities, offering voice, video, and text chat features."
script = "communication-apps/discord-setup.sh"
task_list = "installation"

[[data.entries]]
name = "Jitsi"
description = "Jitsi Meet is an open-source video conferencing tool that allows users to host and join secure meetings directly from their web browsers without the need for downloads."
script = "communication-apps/jitsi-setup.sh"
task_list = "installation"

[[data.entries]]
name = "Signal"
description = "Signal is a privacy-focused messaging app that provides end-to-end encryption for secure text, voice, and video communication."
script = "communication-apps/signal-setup.sh"
task_list = "installation"

[[data.entries]]
name = "Slack"
description = "Slack is a collaboration platform designed for team communication, featuring channels, direct messaging, file sharing, and integrations with various productivity tools."
script = "communication-apps/slack-setup.sh"
task_list = "installation"

[[data.entries]]
name = "Telegram"
description = "Telegram is a cloud-based messaging app known for its speed and security, offering features like group chats, channels, and end-to-end encrypted calls."
script = "communication-apps/telegram-setup.sh"
task_list = "installation"

[[data.entries]]
name = "Zoom"
description = "Zoom is a widely-used video conferencing platform that allows users to host virtual meetings, webinars, and online collaboration with features like screen sharing and recording."
script = "communication-apps/zoom-setup.sh"
task_list = "installation"

[[data.entries]]
name = "Thunderbird"
```

**Review by adamperkowski — CHANGES_REQUESTED** [review]

**Review by adamperkowski — APPROVED** [review]

**Review by adamperkowski — APPROVED** [review]

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**